From 30aa483a5d6e05478e1a1593e07fed2e88346800 Mon Sep 17 00:00:00 2001
From: Guilhem Saurel <guilhem.saurel@laas.fr>
Date: Thu, 22 Oct 2020 13:38:12 +0200
Subject: [PATCH] fix for 5.0 < Qt < 5.9

QTemporaryDir::errorString was introduced in Qt 5.6
QTemporaryDir::filePath was introduced in Qt 5.9

Ubuntu 16.04 has Qt 5.5, and Debian Stretch has 5.7
---
 src/gui/osgwidget.cc | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/src/gui/osgwidget.cc b/src/gui/osgwidget.cc
index 3dec28a..5d07b48 100644
--- src/gui/osgwidget.cc
+++ src/gui/osgwidget.cc
@@ -178,13 +178,24 @@ namespace gepetto {
         }
         tmpDirectory_ = new QTemporaryDir (QDir::temp().absoluteFilePath("gepetto-gui."));
         if (!tmpDirectory_->isValid()) {
-          main->logError(tmpDirectory_->errorString());
+          main->logError(
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 6, 0))
+              tmpDirectory_->errorString()
+#else
+              "A temporary directory is not valid"
+#endif
+          );
           delete tmpDirectory_;
           tmpDirectory_ = NULL;
           recordMovie_->setChecked(false);
         } else {
           tmpDirectory_->setAutoRemove(false);
-          QString path = tmpDirectory_->filePath("img");
+          QString path =
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 9, 0))
+              tmpDirectory_->filePath("img");
+#else
+              tmpDirectory_->path() + "/img";
+#endif
           const char* ext = "jpeg";
           osg ()->startCapture(windowID(), path.toLocal8Bit().data(), ext);
           main->log("Saving images to " + path + "_<number>." + ext);
@@ -202,7 +213,12 @@ namespace gepetto {
       } else {
 #if (QT_VERSION >= QT_VERSION_CHECK(5,0,0))
         if (tmpDirectory_ == NULL) return;
-        QString input (tmpDirectory_->filePath("img_%d.jpeg"));
+        QString input =
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 9, 0))
+            tmpDirectory_->filePath("img_%d.jpeg");
+#else
+            tmpDirectory_->path() + "/img_%d.jpeg";
+#endif
 #else
         QString input = "/tmp/gepetto-gui/record/img_0_%d.jpeg";
 #endif
